x-pack/plugins/cloud_security_posture/public/pages/configurations/latest_findings/latest_findings_container.tsx

```diff
@@ -107,12 +117,229 @@ const customCellRenderer = (rows: DataTableRecord[]) => ({
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByRuleName', {
+      defaultMessage: 'Rule name',
+    }),
+    key: 'rule.name',
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByCloudAccount', {
+      defaultMessage: 'Cloud account',
+    }),
+    key: 'cloud.account.name',
+  },
+  {
+    label: i18n.translate('xpack.csp.findings.latestFindings.groupByKubernetesCluster', {
+      defaultMessage: 'Kubernetes cluster',
+    }),
+    key: 'orchestrator.cluster.name',
+  },
+];
+
+const groupingTitle = i18n.translate('xpack.csp.findings.latestFindings.groupBy', {
+  defaultMessage: 'Group findings by',
+});
+
 export const LatestFindingsContainer = ({ dataView }: FindingsBaseProps) => {
+  const getPersistedDefaultQuery = usePersistedQuery(getDefaultQuery);
+  const { urlQuery, setUrlQuery } = useUrlQuery(getPersistedDefaultQuery);
+  const [activePage, setActivePage] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+
+  const { query } = useBaseEsQuery({
+    dataView,
+    filters: urlQuery.filters,
+    query: urlQuery.query,
+  });
+
+  /**
+   * Reset the active page when the filters or query change
+   * This is needed because the active page is not automatically reset when the filters or query change
+   */
+  useEffect(() => {
+    setActivePage(0);
+  }, [urlQuery.filters, urlQuery.query]);
+
+  const grouping = useGrouping({
+    componentProps: {
+      unit: FINDINGS_UNIT,
+    },
+    defaultGroupingOptions,
+    fields: dataView.fields,
+    groupingId: 'cspLatestFindings',
+    maxGroupingLevels: 1,
+    title: groupingTitle,
+    onGroupChange: () => {
+      setActivePage(0);
+    },
+  });
+
+  const selectedGroup = grouping.selectedGroups[0];
+  const isNoneSelected = isNoneGroup(grouping.selectedGroups);
+
+  const uniqueValue = useMemo(() => `${selectedGroup}-${uuid.v4()}`, [selectedGroup]);
+
+  const groupingQuery = getGroupingQuery({
+    additionalFilters: [query],
+    groupByField: selectedGroup,
+    uniqueValue,
+    from: 'now-1y',
+    to: 'now',
+    pageNumber: activePage * pageSize,
+    size: pageSize,
+    sort: [{ _key: { order: 'asc' } }],
+  });
+
+  const { data, isFetching } = useGroupedFindings({
+    query: groupingQuery,
+    enabled: !isNoneSelected,
+  });
+
+  const aggs = useMemo(
+    () => parseGroupingQuery(selectedGroup, uniqueValue, data),
+    [data, selectedGroup, uniqueValue]
+  );
+
+  if (isNoneSelected) {
+    return (
+      <>
+        <FindingsSearchBar dataView={dataView} setQuery={setUrlQuery} loading={isFetching} />
+        <LatestFindingsContainerTable dataView={dataView} groupSelector={grouping.groupSelector} />
+      </>
+    );
+  }
+
+  return (
+    <div
+      css={css`
+        && .euiAccordion__optionalAction {
+          display: none;
+        }
+      `}
+    >
+      <FindingsSearchBar dataView={dataView} setQuery={setUrlQuery} loading={isFetching} />
+      {grouping.getGrouping({
```

Comment:
a bit nitty but i think it might add readability to create the component and return it here as a JSX element instead

Reply:
I also agree, I think I can add a wrapper component and that can also be useful for being reused later by the vulnerabilities table.

Reply:
thanks for the suggestion, I moved it to its own component, and also added a custom hook to init it and get the required parameters so it will be easier to reuse for other pages